pkg/k8shandler/kibana/route.go

```diff
@@ -94,56 +98,42 @@ func (clusterRequest *KibanaRequest) createOrUpdateKibanaRoute() error {
 		return fmt.Errorf("Failure creating Kibana route for %q: %v", cluster.Name, err)
 	}
 
+	return nil
+}
+
+func (clusterRequest *KibanaRequest) createOrUpdateKibanaConsoleLinks() error {
+	cluster := clusterRequest.cluster
+
 	kibanaURL, err := clusterRequest.GetRouteURL("kibana")
 	if err != nil {
 		return err
 	}
 
-	sharedConfig := createSharedConfig(cluster.Namespace, kibanaURL, kibanaURL)
-	utils.AddOwnerRefToObject(sharedConfig, utils.AsOwner(cluster))
-
-	err = clusterRequest.Create(sharedConfig)
-	if err != nil && !errors.IsAlreadyExists(err) {
-		return fmt.Errorf("Failure creating Kibana route shared config: %v", err)
-	}
-
-	sharedRole := NewRole(
-		"sharing-config-reader",
-		cluster.Namespace,
-		NewPolicyRules(
-			NewPolicyRule(
-				[]string{""},
-				[]string{"configmaps"},
-				[]string{"sharing-config"},
-				[]string{"get"},
-			),
-		),
-	)
+	consoleAppLogsLink := NewConsoleLink(kibanaAppConsoleLinkName, cluster.GetNamespace(), kibanaURL)
+	utils.AddOwnerRefToObject(consoleAppLogsLink, utils.AsOwner(cluster))
 
-	utils.AddOwnerRefToObject(sharedRole, utils.AsOwner(clusterRequest.cluster))
+	err = clusterRequest.Create(consoleAppLogsLink)
+	if err != nil {
+		if !errors.IsAlreadyExists(err) {
+			return fmt.Errorf("Failure creating Kibana console app logs link for %q: %v", cluster.Name, err)
+		}
+		if err := clusterRequest.Update(consoleAppLogsLink); err != nil {
```

Comment:
there's a certain pattern to follow when doing an `Update`. you will want to get the existing object and update it if required and then pass it to `clusterRequest.Update`

https://github.com/openshift/elasticsearch-operator/blob/master/pkg/k8shandler/kibana/reconciler.go#L224-L256